src/coreclr/jit/gentree.cpp

```diff
@@ -23320,7 +23320,7 @@ GenTree* Compiler::gtNewSimdShuffleNode(
 #if defined(TARGET_XARCH)
     uint8_t  control   = 0;
     bool     crossLane = false;
-    bool     needsZero = varTypeIsSmallInt(simdBaseType) && (simdSize != 64);
+    bool     needsZero = varTypeIsSmallInt(simdBaseType) && (simdSize <= 16);
```

Comment:
Is this going to be correct for `simdSize == 32` on hardware that is only `AVX2` capable?

or should this instead be something like: `needsZero = false;` and then under the following path, we'd do `needsZero = true`?
```
if ((varTypeIsByte(simdBaseType) && !compOpportunisticallyDependsOn(InstructionSet_AVX512VBMI_VL)) ||
            (varTypeIsShort(simdBaseType) && !compOpportunisticallyDependsOn(InstructionSet_AVX512BW_VL)))
```

Reply:
Basically just wanting to ensure we still have correct handling for AVX2 only hardware and if we need to change this check to account for that in some way, whether its handling it all right here or having a `needsZero = true;` assignment down one of the relevant paths.

Reply:
I had the same doubt but looking at the logic here(https://github.com/DeepakRajendrakumaran/runtime/blob/MRT143/src/coreclr/jit/gentree.cpp#L23396-L23434), we do not have to use the `needsZero `flag  for **AVX2** only hardware when dealing with `small `types since we are returning here without going into the `needsZero `logic

Reply:
Ah, that's right. We're explicitly using `SSSE3` here and so its implicitly handling `needsZero`. 👍